sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -2053,8 +2054,8 @@ class DataFrameSuite extends QueryTest
       val df = Dataset.ofRows(spark, statsPlan)
 
       val logicalRDD = LogicalRDD(
-        df.logicalPlan.output, spark.sparkContext.emptyRDD, Some(df.queryExecution.analyzed),
-        isStreaming = true)(spark)
+        df.logicalPlan.output, spark.sparkContext.emptyRDD[InternalRow], isStreaming = true)(
+        spark, Some(df.queryExecution.optimizedPlan.stats), None)
```

Comment:
Maybe we can call `fromDataset` instead?

Reply:
Unfortunately we can't, as OutputListAwareStatsTestPlan and OutputListAwareConstraintsTestPlan do not have corresponding physical plan. (output partitioning is picked up from physical plan, as same as we do in Dataset.checkpoint)